newrelic/provider.go

```diff
@@ -29,6 +29,11 @@ func Provider() terraform.ResourceProvider {
 				Optional:    true,
 				DefaultFunc: schema.EnvDefaultFunc("NEWRELIC_API_URL", "https://api.newrelic.com/v2"),
 			},
+			"synthetics_api_url": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("NEWRELIC_SYNTHETICS_API_URL", "https://synthetics.newrelic.com/synthetics/api/v3"),
```

Comment:
Not sure our current practice, but the URL string should be a constant somewhere.  Are these exported by the synthetics package?  i.e.

`DefaultFunc: schema.EnvDefaultFunc("NEWRELIC_SYNTHETICS_API_URL", synthetics.BaseURL.DefaultRegion)`

(not sure any of that exists upstream, but would settle for a const in this package for now)